### PR TITLE
Java provider

### LINKED ISSUE #111: Java provider

### PR DESCRIPTION
**Pull request by wyzlle**
closes #111

### CODE REVIEW

File: src/providers/java.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+pub struct JavaProvider {}
+
+impl Provider for JavaProvider {
+    fn name(&self) -> &str {
+        "Java"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("pom.xml")
```
Comment (Milo123459) [inline src/providers/java.rs:17]: I thought scala, groovy, clojure etc were going in different providers?

Comment (wyzlle) [reply src/providers/java.rs:17]: WIP, will remove them later.

Comment (aleksrutins) [reply src/providers/java.rs:17]: Also, Gradle should probably be supported.

Comment (wyzlle) [reply src/providers/java.rs:17]: I am basically rewriting [heroku's java buildpack](https://github.com/heroku/heroku-buildpack-java). Will add/remove stuff later.

Comment (JakeCooper) [reply src/providers/java.rs:17]: > I thought scala, groovy, clojure etc were going in different providers?

We're gonna do em as one for now and then work on composition once we get to parity

Comment (Milo123459) [reply src/providers/java.rs:17]: May I ask why?

File: examples/java/src/main/resources/banner.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+  _    _                _
+ | |  | |              | |
+ | |__| | ___ _ __ ___ | | ___   _
+ |  __  |/ _ \ '__/ _ \| |/ / | | |
+ | |  | |  __/ | | (_) |   <| |_| |
+ |_|  |_|\___|_|  \___/|_|\_\\__,_|
+
+:: Built with Spring Boot :: ${spring-boot.version}
```
Comment (JakeCooper) [inline examples/java/src/main/resources/banner.txt:8]: Can we make it Railway ascii?

Comment (wyzlle) [reply examples/java/src/main/resources/banner.txt:8]: Yeah, obviously. Will update example.

File: src/providers/java.rs (hunk 1 above)
Comment (JakeCooper) [reply src/providers/java.rs:17]: Simplicity sake for now. Easier to break them up later once we know what the composition looks like than try to define abstractions across providers without knowing what should be composable.

**Review by JakeCooper — APPROVED** [review]

**Review by coffee-cup — APPROVED** [review]

**Pull request merged**